azurerm/internal/services/iothub/iothub_resource.go

```diff
@@ -346,6 +346,38 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"enrichment": {
```

Comment:
Should this be part of the routing block?

Reply:
Do you mean the [route block](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/iothub/iothub_resource.go#L301)

You can have no custom routes and an enrichment for built-in endpoint events. So to me, it has to be seperated